goship.go

```diff
@@ -760,23 +778,210 @@ func extractOutputHandler(fn func(http.ResponseWriter, *http.Request, string, st
 	}
 }
 
+type User struct {
+	UserName, UserAvatar string
+	auth                 auth
+}
+
+func getUser(r *http.Request) (User, error) {
+	u := User{}
+	session, err := store.Get(r, session_name)
+	if err != nil {
+		return u, errors.New("Error Getting User Session")
+	}
+	if _, ok := session.Values["userName"]; !ok {
+		u.UserName = "genericUSER"
+		u.UserAvatar = "https://camo.githubusercontent.com/33a7d9a138ac73ece82dee977c216eb13dffc984/687474703a2f2f692e696d6775722e636f6d2f524c766b486b612e706e67"
+		return u, errors.New("No username")
+	}
+	if _, ok := session.Values["avatarURL"]; !ok {
+		return u, errors.New("No avatar")
+	}
+	// Check return of nil
+	u.UserName = session.Values["userName"].(string)
+	u.UserAvatar = session.Values["avatarURL"].(string)
+	return u, nil
+}
+
+func checkAuth(fn http.HandlerFunc, a auth) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := getUser(r)
+		if err != nil && a.authorization != false {
+			log.Printf("error getting a logged in user %s", err)
+			http.Redirect(w, r, os.Getenv("GITHUB_CALLBACK_URL")+"/auth/github/login", 301)
+
+			return
+		}
+		fn.ServeHTTP(w, r)
+	})
+}
+
+type cleanProject []goship.Project
+
+// remove projects where user has no access
+func cleanProjects(n cleanProject, r *http.Request, u User) cleanProject {
+	f_projects := []goship.Project{}
```

Comment:
It's not clear to me what the `f` in `f_projects` is for here? Also, can we keep to the camelCase convention here too :bow: